include/vigra/mathutil.hxx

```diff
@@ -164,6 +164,17 @@ VIGRA_DEFINE_MISSING_ABS(signed long long)
 using std::isinf;
 using std::isnan;
 
+#ifdef _LIBCPP_VERSION
+// On older versions of Mac, the system's libc++ does not provide isnan(double)
+// So we just provide it ourselves.
```

Comment:
Is this addition safe? I know we have isnan() workarounds in other places already, so we should probably agree how to do this in vigra?! For instance, I used `#include <boost/math/special_functions/fpclassify.hpp>` and then `boost::math::isnan` for maximum compatibilty. (I see that `isnan()` may be simple enough not to require boost for that if it is not already a requirement.)